core/templates/dev/head/domain/utilities/LanguageUtilServiceSpec.js

```diff
@@ -67,4 +76,33 @@ describe('Language util service', function() {
     expect(lus.getComplementAudioLanguageCodes(['abcdefg'])).toEqual([
       'en', 'hi-en', 'es']);
   });
+
+  it('should correctly get related language code given audio language code',
+    function() {
+      expect(lus.getLanguageCodesRelatedToAudioLanguageCode('en')).
+        toEqual(['en']);
+      expect(lus.getLanguageCodesRelatedToAudioLanguageCode('hi-en')).
+        toEqual(['hi', 'en']);
+      expect(lus.getLanguageCodesRelatedToAudioLanguageCode('es')).
+        toEqual(['es']);
+    });
+
+  it('should correctly check if language supports autogenerated audio',
+    function() {
+      expect(lus.supportsAutogeneratedAudio('en')).toEqual(false);
+      // Selenium browser doest not support speech synthesis functionality yet
+      // Expecting this to be true once Selenium provides speech synthesis
+      expect(lus.supportsAutogeneratedAudio('en-auto')).toEqual(false);
```

Comment:
Can we use provider.value for `BrowserCheckerService.supportsSpeechSynthesis` to return true as of now and test our `AUTOGENERATED_AUDIO_LANGUAGES` list in the given service?
```
$provide.value('BrowserCheckerService', {
     supportsSpeechSynthesis: function() {
        return true;
     }
  });
```
Is there any issue that way? (not sure!)

Reply:
Currently working on making `BrowserCheckerService.supportsSpeechSynthesis` to return `true` organically. Will resort to @DubeySandeep solution as a final solution